### PR TITLE
Make `ChainSelStarvation` carry an `Enclosed`

### PR DESCRIPTION
**Pull request by jasagredo**
The definition of this tracer, carrying a `Time`, was basically isomorphic to carrying an `Enclosed`.

### CODE REVIEW

**Review by Niols — APPROVED** [review]
LGTM.

Note that the previous `ChainSelStarvation*` events carried a time. This was added the first time we saw very short starvations, got extremely, and wanted to be sure to measure their exact length, without potential delays introduced in the tracing system. After discussion with @jasagredo, we decided that it was most likely not necessary at this point; it can always be re-introduced subsequently. The time is of course still written in the `starvationVar`, because that is used by the BlockFetch logic.

**Comment by Niols** [comment]
(Impressive that you stumble about this time limited leashing attack bug again, considering how rare it is! It also makes one wonder how we didn't notice it beforehand.)

File: ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/Types.hs [hunk 1]
```diff
@@ -938,11 +937,10 @@ data TraceIteratorEvent blk
...
 -- them, even though we generally expect to be CPU-bound.
-data TraceChainSelStarvationEvent blk
-    -- | A ChainSel starvation started at the given time.
-  = ChainSelStarvationStarted Time
-
-    -- | The last ChainSel starvation ended at the given time as a block wth the
-    -- given point has been received.
-  | ChainSelStarvationEnded Time (RealPoint blk)
+--
+-- Somehow it happens regularly during syncing for very short times.
```
Comment (nfrisby) [inline ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/Types.hs:941]: > Somehow it happens regularly during syncing for very short times.

This sentence seems a little out of place. Should it be a TODO (investigate why we're seeing these hiccups)? An Issue?

**Review by nfrisby — APPROVED** [review]
The change seems fine, since it seems unlikely we'll truly need more than the centisecond precision the node's logging infrastructure gives on tracer times in the log.

I have no strong preference either way, but I haven't been the one debugging these error traces. So I'll defer to you and Niols.

**Pull request merged**